src/system-health/health_checker/service_checker.py

```diff
@@ -24,23 +43,177 @@ class ServiceChecker(HealthChecker):
+        ctrs = DOCKER_CLIENT.containers
+        try:
+            lst = ctrs.list(filters={"status": "running"})
+
+            for ctr in lst:
+                running_containers.add(ctr.name)
+                if ctr.name not in self.container_critical_processes:
+                    self.fill_critical_process_by_container(ctr.name)
+        except docker.errors.APIError as err:
+            logger.log_debug("Failed to retrieve the running container list. Error: '{}'".format(err))
+
+        return running_containers
+
+    def get_critical_process_list_from_file(self, container, critical_processes_file):
+        """Read critical process name list from critical processes file
+
+        Args:
+            container (str): contianer name
+            critical_processes_file (str): critical processes file path
+
+        Returns:
+            critical_process_list: A list of critical process names
+        """
+        critical_process_list = []
+
+        with open(critical_processes_file, 'r') as file:
+            for line in file:
+                # Try to match a line like "program:<process_name>"
+                match = re.match(r"^\s*((.+):(.*))*\s*$", line)
+                if match is None:
+                    if container not in self.bad_containers:
+                        self.bad_containers.add(container)
+                        logger.log_error('Invalid syntax in critical_processes file of {}'.format(container))
+                    continue
+
+                identifier_key = match.group(2).strip()
+                identifier_value = match.group(3).strip()
+                if identifier_key == "program" and identifier_value:
+                    critical_process_list.append(identifier_value)
+
+        return critical_process_list
+
+    def fill_critical_process_by_container(self, container):
+        """Get critical process for a given container
+
+        Args:
+            container (str): container name
+        """
+        # Get container volumn folder
+        container_folder = self._get_container_folder(container)
+        if not container_folder:
+            logger.log_debug('Failed to get container folder for {}'.format(container_folder))
+            return
+
+        # If container folder does not exist, the container is probably not up, retry it
+        if not os.path.exists(container_folder):
+            logger.log_debug('Container folder does not exist: {}'.format(container_folder))
+            return
+
+        # Get critical_processes file path
+        critical_processes_file = os.path.join(container_folder, ServiceChecker.CRITICAL_PROCESSES_PATH)
+        if not os.path.isfile(critical_processes_file):
+            # Critical process file does not exist, the container has no critical processes.
+            # This is fine, don't retry.
+            logger.log_debug('Failed to get critical process file for {}, {} does not exist'.format(container, critical_processes_file))
```

Comment:
`log_debug(...)` to `log_error(...)`?

Reply:
Debug log is fine here. Some container might not have critical processes file, this is allowed.